### PR TITLE
Improve error message on storage tags update

### PR DESCRIPTION
**Pull request by BryanMLima**
### Description

This PR aims to improve the error message while trying to update the storage tags of a Disk Offering when none of the Primary Storages have that set of storage tags. The current behavior only showed the ID of the Disk Offering and Storage Pools. This PR improves the error message to display the UUID of these components, as well as their names and the volumes using that Disk Offering. Furthermore, it was added unit tests for the modified method `updateOfferingTagsIfIsNotNull`.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

### Screenshots (if appropriate):
image

### How Has This Been Tested?
This was tested in a local lab trying to update the storage tags of a disk offering, when none of the primary storages had that set of storage tags. The error message returned is in the Screenshots section.

### CODE REVIEW

File: engine/schema/src/main/java/com/cloud/storage/dao/VolumeDao.java [hunk 1]
```diff
@@ -131,4 +131,12 @@ public interface VolumeDao extends GenericDao<VolumeVO, Long>, StateDao<Volume.S
      *  Updates the disk offering for the given volume.
      */
     void updateDiskOffering(long volumeId, long diskOfferingId);
+
+    /**
+     *  Get volumes by the disk offering ID.
```
Comment (GutoVeronezi) [inline engine/schema/src/main/java/com/cloud/storage/dao/VolumeDao.java:136]:
```suggestion
     *  Gets volumes by the disk offering ID.
```

File: server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 2]
```diff
@@ -3801,7 +3803,15 @@ protected void updateOfferingTagsIfIsNotNull(String tags, DiskOfferingVO diskOff
...
                     List<String> tagsOnPool = storagePoolTagDao.getStoragePoolTags(storagePoolVO.getId());
                     if (CollectionUtils.isEmpty(tagsOnPool) || !tagsOnPool.containsAll(listOfTags)) {
-                        throw new InvalidParameterValueException(String.format("There are active volumes using offering [%s], and the pools [%s] don't have the new tags", diskOffering.getId(), pools));
+                                                String offeringUuid = _diskOfferingDao.findById(diskOffering.getId()).getUuid();
+                        List<VolumeVO> volumes = _volumeDao.findByDiskOfferingId(diskOffering.getId());
+                        List<String> listOfVolumesNamesAndUuid = new ArrayList<>();
+                        for (VolumeVO volumeVO : volumes) {
+                            listOfVolumesNamesAndUuid.add(String.format("Name: %s, UUID: %s",volumeVO.getName(), volumeVO.getUuid()));
+                        }
+                        throw new InvalidParameterValueException(String.format("There are active volumes using the disk offering [Name: %s, UUID: %s], and the pool [Name: %s, UUID: %s] doesn't have the new tags. " +
```
Comment (GutoVeronezi) [inline server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java:3812]: We can use `ReflectionToStringBuilderUtils` here.

**Review by GabrielBrascher — APPROVED** [review]
Code LGTM.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

**Comment by GutoVeronezi** [comment]
@DaanHoogland, can we try to run the tests again?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
I doubt these errors have anything to do with the change. I'll try a re-run again
@blueorangutan test

**Comment by DaanHoogland** [comment]
note, added milestone 4.19.0.0 in retrospect.

**Pull request merged**